packages/jaeger-ui/src/components/TracePage/Theme.tsx

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React from 'react';
+import hoistNonReactStatics from 'hoist-non-react-statics';
+import memoizeOne from 'memoize-one';
+
+export type ThemeOptions = Partial<Theme>;
+
+export type Theme = {
```

Comment:
I think this type would make more sense if the attributes were scoped to the component they were styling. Something like:
```
export type Theme = Record<string, Partial<emotion-style-object-type>>
```
(I assume emotion has a type for valid css objects)

then theme consumers can have different `borderStyle`s or what have you based on what is being styled, e.g.:
```
const theme: TTheme = {
  TraceTimelineViewer: {
    borderBottom: '1px solid #bbb',
  },
}
```

also types in this codebase should be prefixed by a `T` (enums get `E`, and interfaces get `I`)